From ebaea8ed74fc8824760911332ac7b9e68ec0ad15 Mon Sep 17 00:00:00 2001
From: tobitege <tobiasteschner@googlemail.com>
Date: Mon, 18 May 2015 10:43:39 +0200
Subject: [PATCH 042/296] Fix NPE in updateEmptyShadeView() upon screen
 rotation

FATAL EXCEPTION: main
Process: com.android.systemui, PID: 19252

java.lang.NullPointerException: Attempt to invoke virtual method
'boolean com.android.systemui.statusbar.EmptyShadeView.willBeGone()' on a null object reference
  at com.android.systemui.statusbar.stack.NotificationStackScrollLayout.updateEmptyShadeView(NotificationStackScrollLayout.java:2291)
  at com.android.systemui.statusbar.phone.NotificationPanelView.updateEmptyShadeView(NotificationPanelView.java:2058)
  at com.android.systemui.statusbar.phone.NotificationPanelView.updateQsState(NotificationPanelView.java:1181)
  at com.android.systemui.statusbar.phone.NotificationPanelView.setQsScrimEnabled(NotificationPanelView.java:2065)
  at com.android.systemui.statusbar.phone.PhoneStatusBar$11.onVerticalChanged(PhoneStatusBar.java:1079)
---
 .../android/systemui/statusbar/stack/NotificationStackScrollLayout.java  | 1 +
 1 file changed, 1 insertion(+)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/stack/NotificationStackScrollLayout.java b/packages/SystemUI/src/com/android/systemui/statusbar/stack/NotificationStackScrollLayout.java
index 75532d9e09b..eb263906328 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/stack/NotificationStackScrollLayout.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/stack/NotificationStackScrollLayout.java
@@ -3834,6 +3834,7 @@ public class NotificationStackScrollLayout extends ViewGroup
     }
 
     public void updateEmptyShadeView(boolean visible) {
+        if (mEmptyShadeView == null) return;
         int oldVisibility = mEmptyShadeView.willBeGone() ? GONE : mEmptyShadeView.getVisibility();
         int newVisibility = visible ? VISIBLE : GONE;
         if (oldVisibility != newVisibility) {
-- 
2.11.0

